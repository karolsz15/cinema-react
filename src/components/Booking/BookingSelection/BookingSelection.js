import React from 'react';
import PropTypes from 'prop-types';
import { format, add } from 'date-fns';

const bookingSelection = (props) => {
  const { changed } = props;

  const fullDate = new Date();
  const dayOfMonth = format(fullDate, 'dd');
  const dayOfWeek = format(fullDate, 'cccc');
  const month = format(fullDate, 'LL');

  String.prototype.unCapitalize = function () {
    return this.charAt(0).toLowerCase() + this.slice(1);
  };

  return (
    <select
      onChange={changed}
      defaultValue={dayOfWeek}
      className="custom-select custom-select-lg mb-4"
      id="day"
      name="daySelection"
    >
      {/* default value - current day */}
      <option value={dayOfWeek.unCapitalize()}>
        {dayOfWeek} {dayOfMonth}.{month}
      </option>

      {/* options' values - list of future six days */}
      {[1, 2, 3, 4, 5, 6].map((n) => (
        <option
          value={format(add(fullDate, { days: n }), 'cccc').unCapitalize()}
          key={format(add(fullDate, { days: n }), 'cccc')}
        >
          {format(add(fullDate, { days: n }), 'cccc')} {format(add(fullDate, { days: n }), 'dd')}.
          {format(add(fullDate, { days: n }), 'LL')}
        </option>
      ))}
    </select>
  );
};

bookingSelection.propTypes = {
  changed: PropTypes.func,
};

export default bookingSelection;
