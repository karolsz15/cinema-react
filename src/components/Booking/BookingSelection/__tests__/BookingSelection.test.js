import React from 'react';
import { shallow } from 'enzyme';
import BookingSelection from '../BookingSelection';

let wrapper = shallow(<BookingSelection />);

describe('BookingSelection component', () => {
  it('should render the BookingSelection correctly', () => {   
    expect(wrapper).toMatchSnapshot();
  });
});