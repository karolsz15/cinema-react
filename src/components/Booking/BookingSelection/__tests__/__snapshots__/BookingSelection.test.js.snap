// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookingSelection component should render the BookingSelection correctly 1`] = `
<select
  className="custom-select custom-select-lg mb-4"
  defaultValue="Friday"
  id="day"
  name="daySelection"
>
  <option
    value="friday"
  >
    Friday
     
    25
    .
    02
  </option>
  <option
    key="Saturday"
    value="saturday"
  >
    Saturday
     
    26
    .
    02
  </option>
  <option
    key="Sunday"
    value="sunday"
  >
    Sunday
     
    27
    .
    02
  </option>
  <option
    key="Monday"
    value="monday"
  >
    Monday
     
    28
    .
    02
  </option>
  <option
    key="Tuesday"
    value="tuesday"
  >
    Tuesday
     
    01
    .
    03
  </option>
  <option
    key="Wednesday"
    value="wednesday"
  >
    Wednesday
     
    02
    .
    03
  </option>
  <option
    key="Thursday"
    value="thursday"
  >
    Thursday
     
    03
    .
    03
  </option>
</select>
`;
